src/llmcompressor/pipelines/sequential/helpers.py

```diff
@@ -0,0 +1,365 @@
+            if isinstance(a, PretrainedConfig):
+                kwargs = {k: self.create_arg(v) for k, v in a.to_dict().items()}
+                return self.create_node("call_function", a.__class__, (), kwargs)
+
+            else:
+                return super().create_arg(a)
+
+        def is_leaf_module(self, module: Module, module_qualified_name: str) -> bool:
+            return module in skip_trace_modules or super().is_leaf_module(
+                module, module_qualified_name
+            )
+
+    return SequentialTracer()
+
+
+def populate_concrete_args(model: Module, sample_input: Dict) -> Dict:
+    """
+    Creates concrete args which, unlike the equivalent function provided by
+    transformers.utils.fx, creates default values for variadic arguments, which are
+    needed by some models.
+
+    :param model: model being traced
+    :param sample_input: values used to symbolically trace the model. All arguments
+        to the model.forward function which are not in the sample_input are considered
+        concrete args
+    :return: dictionary mapping concrete argument names to their default values
+    """
+    sig = inspect.signature(model.forward)
+
+    concrete_args = {}
+    for parameter in sig.parameters.values():
+        if parameter.name in sample_input:
+            continue
+        if parameter.kind == inspect._ParameterKind.VAR_POSITIONAL:
+            value = list()
+        elif parameter.kind == inspect._ParameterKind.VAR_KEYWORD:
+            value = dict()
+        elif parameter.name == "use_cache":
+            value = False
+        else:
+            value = parameter.default
+
+        concrete_args[parameter.name] = value
+
+    return concrete_args
+
+
+def find_target_nodes(graph: GraphModule, targets: Set[Module]) -> Set[Node]:
+    """
+    Find all nodes whose execution is equivalent to executing the target modules.
+    Note that these nodes are guaranteed to be treated as leaf nodes by SequentialTracer
+
+    :param graph: graph containing target nodes
+    :param targets: modules whose nodes are being searched for
+    :return: set of all nodes which call the target modules
+    """
+    return set(
+        node
+        for node in graph.graph.nodes
+        if node.op == "call_module" and graph.get_submodule(node.target) in targets
+    )
+
+
+def topological_partition(graph: GraphModule, targets: Set[Module]) -> List[List[Node]]:
+    """
+    Partition the graph into partitions such that each `target` belongs to exactly one
+    partition and executing each partition depends only on intermediate values produced
+    by executing the partitions before it.
+
+    :param graph: graph being partitioned
+    :param targets: target modules which will be assigned to disjoint partitions
+    :return: list of partitions, where each partition is a list of nodes belonging to
+        that partition
+    """
+    assert check_assumption(graph.graph)
```

Comment:
nit: update for a more descriptive name e.g check_graph_not_malformed or something